### PR TITLE
Implement slash option min/max length

### PR DESCRIPTION
**Pull request by HyperGH**
### Summary
Add `min_length` and `max_length` to slash command options. For more info see: https://discord.com/developers/docs/change-log#min-and-max-length-for-command-options

### Checklist

- [x] I have run `nox` and all the pipelines have passed.
- [x] I have made unittests according to the code I have added/modified/deleted.

### CODE REVIEW

File: tests/hikari/internal/test_cache.py [hunk 1]
```diff
@@ -20,6 +20,7 @@
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
 import copy
+
```
Comment (davfsa) [inline tests/hikari/internal/test_cache.py:23]: Random space

**Review by davfsa — APPROVED** [review]
LGTM, just a random thing that was probably added by mistake

**Comment by tandemdude** [comment]
epic pr now i add more params to option decorator very good very nice

**Comment by Jonxslays** [comment]
Needs test cases as well.

**Review by davfsa — APPROVED** [review]

**Review by FasterSpeeding — APPROVED** [review]

**Pull request merged**